File: Week1/C/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int convertToCents(float amount);
void minimumRequiredCoins(int changeDue);

int main(void)
{
    float amount; //stores the amount due to the customer
    
    do
    {
        amount = get_float("Change owed: ");
    }
    while (amount < 0.00); //checks for any invalid negative amount asked

    int changeDue = convertToCents(amount); //stores the equivalent dollar amount in cents
    minimumRequiredCoins(changeDue); //finds the minimum coins required to pay the amount owed
}

int convertToCents(float amount)
{
    return round(amount * 100); //returns the cents equivalent of the asked amount
}

void minimumRequiredCoins(int changeDue)
{
    //coins stores the individual cents that is required, totalCoins stores the minimum total coins needed
    int totalCoins = 0, coins = 0;

    while (changeDue >= 5)
    {
        if (changeDue / 25 > 0)
        {
            coins = changeDue / 25;
            changeDue -= coins * 25;
            totalCoins += coins;
        }
        else if (changeDue / 10 > 0)
        {
            coins = changeDue / 10;
            changeDue -= coins * 10;
            totalCoins += coins;
        }
        else if (changeDue / 5 > 0)
        {
            coins = changeDue / 5;
            changeDue -= coins * 5;
            totalCoins += coins;
        }
    }
    totalCoins += changeDue; //adds all the remaining 1 cents needed
    printf("%d\n", totalCoins);
}

File: Week6/Python/cash.py
from cs50 import get_float


def main():
    while True:
        amount = get_float("Change owed: ")
        # checks for correct input condition
        if amount >= 0.00:
            break
    # stores the equivalent dollar amount in cents
    changeDue = converttoCents(amount)
    # finds the minimum coins required to pay the amount owed
    minimumRequiredCoins(changeDue)


def converttoCents(amt):
    # returns the cents equivalent of the asked amount
    return round(amt * 100)


def minimumRequiredCoins(amt):
    # coins stores the individual cents that is required, totalCoins stores the minimum total coins needed
    totalCoins = 0

    while amt >= 5:

        if (amt // 25) > 0:
            coins = amt // 25
            amt -= coins * 25

        elif (amt // 10) > 0:
            coins = amt // 10
            amt -= coins * 10

        elif (amt // 5) > 0:
            coins = amt // 5
            amt -= coins * 5

        totalCoins += coins

    # adds all the remaining 1 cents needed
    totalCoins += amt

    print(f"{totalCoins}")


main()

